cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataKey.java

```diff
@@ -141,6 +163,11 @@ private static MetadataEntity getTargetIdIdFromKey(MDSKey.Splitter keySplitter)
     builder.add(metadataEntity.getType());
     // add all the key value pairs from the metadata entity this is the targetId
     for (MetadataEntity.KeyValue keyValue : metadataEntity) {
+      // CDAP-13597 for versioned entities like application, schedule and programs their metadata is not versioned
```

Comment:
is this behavior documented? Users could get confused if it is not (and even if it is)

Reply:
https://issues.cask.co/browse/CDAP-13611